wolfcrypt/test/test.c

```diff
@@ -10343,6 +10350,40 @@ WOLFSSL_TEST_SUBROUTINE int aesccm_test(void)
     if (XMEMCMP(pl, pl2, sizeof(pl2)))
         ERROR_OUT(-6520, out);
 
+    /* test empty message as null input and output --
```

Comment:
There are some implementations that don't allow NULL and 0.
Need to exclude these tests in those cased.

Reply:
Ah, those cases are already excluded, by tolerating `BAD_FUNC_ARG`, but _only_ `BAD_FUNC_ARG`, from `wc_AesCcmEncrypt(enc, NULL /* out */, NULL /* in */, 0 /* inSz */, ...)`.

Iff the `wc_AesCcmEncrypt(enc, NULL, ...)` succeeds, then the `wc_AesCcmDecrypt(enc, NULL, ...)` must succeed.

Note that I also added new empty-message encode-decode cycle that must pass on all targets, that pass as inputs explicit non-NULL pointers to `""`, and non-null output buffers.

Reply:
I've also just added tests to make sure null pointers with `inSz > 0` always return `BAD_FUNC_ARG`.